File: src/main/java/org/kite9/diagram/visualization/compaction/CompactionStep.java
package org.kite9.diagram.visualization.compaction;

import org.kite9.diagram.model.Rectangular;

/**
 * A step in which some part of the compaction process occurs.
 * 
 * 
 * @author robmoffat
 *
 */
public interface CompactionStep {

	public void compact(Compaction c, Rectangular r, Compactor rc);
	
}

File: kite9-visualization/src/common/kotlin/org/kite9/diagram/visualization/compaction/CompactionStep.kt
package org.kite9.diagram.visualization.compaction

/**
 * A step in which some part of the compaction process occurs.
 *
 *
 * @author robmoffat
 */
interface CompactionStep {

    fun compact(c: Compaction, e: Embedding, rc: Compactor)
}